scalafmt-macros/shared/src/main/scala/org/scalafmt/config/DialectMacro.scala

```diff
@@ -0,0 +1,33 @@
+package org.scalafmt.config
+
+import scala.meta.Dialect
+
+import scala.language.experimental.macros
+import scala.reflect.macros.blackbox
+
+// Builds a map between string and dialect method application
+object DialectMacro {
+  def dialectMap(dialect: Dialect): Map[String, (Any => Dialect)] =
```

Comment:
do you really need the dialect here? is it possible to compute instead, and once, `Map[String, (Dialect, Any) => Dialect]`?

Reply:
To be clear, this is also computed once, the call with `dialect` gets expanded once on compiletime to a Map that references the dialect variable, so it ends up looking something like this:
```scala
val map = Map(("withAllowSignificantIndentation", (v: Any) => dialect.withAllowSignificantIndentation(v)),...)
```
But I'll still try to change this to Map[String, (Dialect, Any) => Dialect]